src/Model/RouteItem/BaseRouteItem.php

```diff
@@ -25,13 +26,15 @@ abstract class BaseRouteItem
     protected ParameterBag $routeParameters;
     protected ParameterBag $routeAttributes;
     protected ParameterBag $routeContext;
+    protected ?UrlSlug $urlSlug = null;
```

Comment:
we should not add new properties to the route item which are very specific!

why not using the route context bag?

```php
$routeItem->getRouteContextBag()->set('urlSlug', $urlSlug);
```